src/Model/UserAction.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\core\DIContainerTrait;
+use atk4\core\Exception;
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+use atk4\data\Model;
+
+/**
+ * Implements generic user action. Assigned to a model it can be invoked by a user. Model\UserAction class contains a
+ * meta information about the action (arguments, permissions, scope, etc) that will help UI/API or add-ons to display
+ * action trigger (button) correctly in an automated way.
+ *
+ * UserAction must NOT rely on any specific UI implementation.
+ *
+ * @property Model $owner
+ */
+class UserAction
```

Comment:
(not strictly needed to be a requirement for this PR)

UserAction is related with ui only

Do you think we can easily move everything related with UserAction from data?

Reply:
It can be moved but I believe place is here as UserActions can be used for an API for instance (without UI) so having them under atk4/data is good IMHO.

Reply:
Makes sense to some sense, but then it should be named `Action` instead of `UserAction`.

But the current impl. is very oriented as beiing a helper for ui, see https://github.com/atk4/data/blob/develop/src/Model.php#L431 (there should be no use of `ui` in `data`, hit this several times when I was fixing some issues).

Reply:
That is why I called it Action first. But API also has an User so it is again UserAction.
Let's keep it here and just merge the NS refactor. Now the property name is changed as well.
Further changes can be discussed on Discord with everybody

Reply:
yes